src/topics/inputs.md

```diff
@@ -143,13 +143,56 @@ The `baseCommand` field will always appear in the final command line before the
 
 [touch]: http://www.linfo.org/touch.html
 
+### Optional Input Parameters
+
+Optional input parameters must include `label` and `secondaryFiles`.
+`label` is a short, human-readable description for the parameter.
+
+```cwl
+inputs:
+  example_file:
+    type: File
+    label: Use case for label parameter
+    inputBinding:
+      position: 1
+```
+
+`secondaryFiles` is an optional input parameter that provides a pattern of specifying files
+or directories that must be included alongside the primary file. 
+The following example demonstrates the `secondaryFiles` input parameter.
+
+```cwl
+inputs:
+  example_file:
+    type: File
+    secondaryFiles: [example_file.txt]
+```
+
+Also, a file object listed in `secondaryFiles` may contain nested `secondaryFiles` as shown below:
+
+```{code-block} cwl
+inputs:
+  example_file:
+    type: File
+    secondaryFiles: [
+            {
+                example_file.txt
+                secondaryFiles: [example_file_2.txt]
+            }
+        ]
+```
+
+Note that secondary files are only valid when `type` has a value `File`, or is an array of `items: File`.
+All listed secondary files must be present in the same directory as the primary file,
+since an implementation may fail workflow execution if a listed secondary file is not present. 
+
 ## Array Inputs
 
 It is easy to add arrays of input parameters represented to the command
 line. There are two ways to specify an array parameter. First is to provide
 `type` field with `type: array` and `items` defining the valid data types
 that may appear in the array. Alternatively, brackets `[]` may be added after
-the type name to indicate that input parameter is array of that type.
+the `type: name` to indicate that the input parameter is an array of that type.
```

Comment:
The first change on this line is incorrect.

"type name" means the name of the type (the string following the 'type:' field) not literally `type: name`